eventlet/patcher.py

```diff
@@ -313,10 +315,9 @@ def monkey_patch(**on):
                 if hasattr(orig_mod, attr_name):
                     delattr(orig_mod, attr_name)
 
-            _os = original('os')
-            if name == 'threading' and hasattr(_os, 'register_at_fork'):
+            if name == 'threading' and hasattr(os, 'register_at_fork'):
                 def fix_threading_active(
-                    _global_dict=original('threading').current_thread.__globals__,
+                    _global_dict=_threading.current_thread.__globals__,
                     _patched=orig_mod
```

Comment:
@tipabu it's called patched but refers to pseudo-original module (that is original at first `monkey_patch` call but patched at second). Which module `._active` should be used?

Reply:
Sorry, I think this was born out of confusion -- honestly, I'm probably *still* a fair bit confused. IIRC, the discussion at #592 made it seem that the core problem was the original threading's `current_thread` having the wrong notion of what the active threads were, and we needed to replace it with the patched active threads. So I waited until *after* `orig_thread` becomes the patched version (i.e., after all the `setattr`s), and that's why I named it as I did.

I'm also a little confused about the options you presented: "original" and "patched" make sense enough, but what do you mean by "current but with updated name"?

What's **really** confusing me now, though, is that I can't seem to reproduce the original problem. I comment out the call to `fix_threading_active` and my test still passes :confused: I *know* I tested that it would fail at the time. I also know I've upgraded my Python since then (I'm currently on 3.7.9 and 3.8.6 which were only released in August and September) -- maybe some fix made its way in upstream?

Reply:
@tipabu so it should be patched, as named. Thank you, clear now.

Third option is related to `monkey_patch` getting "original" from `sys.modules`, it would actually contain patched modules if `monkey_patch` was ever to be run twice.

I believe you did test, that's frustrating to not reproduce, yeah. Even if there is upstream fix, we can't force users to upgrade Python.